include/hipSYCL/compiler/Frontend.hpp

```diff
@@ -224,7 +224,7 @@ class FrontendASTVisitor : public clang::RecursiveASTVisitor<FrontendASTVisitor>
     // Construct unique name mangler if supported
     NameMangler = clang::ItaniumMangleContext::create(
       instance.getASTContext(), instance.getASTContext().getDiagnostics(), true);
-#elif  LLVM_VERSION_MAJOR < 13
+#elif  LLVM_VERSION_MAJOR < 13 || defined(HIPSYCL_NO_DEVICE_MANGLER)
```

Comment:
You also need to make sure to handle the case in `nameKernel()`,  otherwise you access a nullpointer there in explicit multipass scenario if the DeviceMangler is not initialized!

Reply:
Ohh thanks! fixed.